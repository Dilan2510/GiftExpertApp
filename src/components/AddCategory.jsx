import PropTypes from "prop-types";
import { useState } from "react";

export const AddCategoryApp = ({ onNewCategory }) => {
  const [input, setInputValue] = useState("");

  const searchInput = (e) => {
    setInputValue(e.target.value);
  };

  const OnSubmit = (event) => {
    event.preventDefault();
    if (input.trim() >= 1) return;
    setInputValue("");
    onNewCategory(input.trim());
  };

  return (
    <form onSubmit={OnSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Buscar Gift"
        value={input}
        onChange={searchInput}
      />
    </form>
  );
};

AddCategoryApp.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
