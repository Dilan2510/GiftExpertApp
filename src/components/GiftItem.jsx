import PropTypes from "prop-types";

export const GiftItem = ({ img, title }) => {
  return (
    <div className="card">
      <img src={img} alt={title} />
      <h5>{title}</h5>
    </div>
  );
};

GiftItem.propTypes = {
  title: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
};
