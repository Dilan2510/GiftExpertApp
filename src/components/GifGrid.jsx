import useFetchGift from "../hooks/useFetchGift";
import { GiftItem } from "./GiftItem";

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGift(category);

  return (
    <div>
      <h3 className="title-card ">{category}</h3>
      {isLoading && <h2>Cargando....</h2>}
      <div className="card-grid">
        {images.map((res) => (
          <GiftItem key={res.id} {...res} />
        ))}
      </div>
    </div>
  );
};
