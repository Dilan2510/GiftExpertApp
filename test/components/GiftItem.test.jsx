import { render, screen } from "@testing-library/react";
import { GiftItem } from "../../src/components/GiftItem";

describe("testing in component GiftItem", () => {
  const name = "Dragon Ball";
  const url =
    "https://media2.giphy.com/media/26CSE3WZrMsVO/giphy.gif?cid=13c727d5ybdj0k9pc7plv75ayd5xu2srkaje6x1mz10n4ci3&ep=v1_gifs_search&rid=giphy.gif&ct=g";
  test("add match with  the Snapshot ", () => {
    const { container } = <GiftItem title={name} img={url} />;
    expect(container).toMatchSnapshot();
  });
  test("testing img from value Equal in const ", () => {
    render(<GiftItem title={name} img={url} />);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(name);
  });
  test("testing in <h5> from title ", () => {
    render(<GiftItem title={name} img={url} />);
    expect(screen.getByRole("heading", { level: 5 }).innerHTML).toBe(name);
  });
});
