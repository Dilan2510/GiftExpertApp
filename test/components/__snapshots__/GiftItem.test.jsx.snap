// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing in component GiftItem add match with  the Snapshot  1`] = `undefined`;
